tests/conftest.py

```diff
@@ -257,3 +257,85 @@ def client(admin: KeycloakAdmin, realm: str) -> str:
     client_id = admin.create_client(payload={"name": client, "clientId": client})
     yield client_id
     admin.delete_client(client_id=client_id)
+
+
+@pytest.fixture
+def client_role(admin: KeycloakAdmin, realm: str, client: str) -> str:
+    """Fixture for a new random client role."""
+    admin.realm_name = realm
+    role = str(uuid.uuid4())
+    admin.create_client_role(client, {"name": role, "composite": False})
+    yield role
+    admin.delete_client_role(client, role)
+
+
+@pytest.fixture
+def composite_client_role(admin: KeycloakAdmin, realm: str, client: str, client_role: str) -> str:
+    """Fixture for a new random composite client role."""
+    admin.realm_name = realm
+    role = str(uuid.uuid4())
+    admin.create_client_role(client, {"name": role, "composite": True})
+    role_repr = admin.get_client_role(client, client_role)
+    admin.add_composite_client_roles_to_role(client, role, roles=[role_repr])
+    yield role
+    admin.delete_client_role(client, role)
+
+
+@pytest.fixture
+def selfsigned_cert():
+    """Generate self signed certificate for a hostname, and optional IP addresses."""
+    import ipaddress
```

Comment:
Yes in general I agree with you, in this case I did it like this since these imports are not used anywhere else, but if you want I can move them outside at the top of the file

Reply:
Right, if that's the only reason, then I'll kindly ask you to move it to the top :)

Reply:
Already done it ;)